src/careamics/lvae_training/dataset/vae_dataset.py

```diff
@@ -461,25 +368,46 @@ def reduce_data(
                 t_list, h_start:h_end, w_start:w_end, :
             ].copy()
 
-        self.N = len(t_list)
         self.set_img_sz(self._img_sz, self._grid_sz)
         print(
             f"[{self.__class__.__name__}] Data reduced. New data shape: {self._data.shape}"
         )
 
+    def get_idx_manager_shapes(self, patch_size: int, grid_size: int):
+        numC = self._data.shape[-1]
+        if self._5Ddata:
+            grid_shape = (1, 1, grid_size, grid_size, numC)
```

Comment:
changes for 3D dataset looks good.  For evaluation, minor changes needs to be done, like grid_size will be a tuple. But that can happen later.

Reply:
Nice, thank you for checking